src/linear_algebra/multilinear.lean

```diff
@@ -1013,6 +1025,125 @@ def multilinear_curry_right_equiv :
+  map_add' := λ u i x y, ext $ λ v,
+    by simp only [multilinear_map.coe_mk, add_apply, ← sum.update_elim_inl, f.map_add],
+  map_smul' := λ u i c x, ext $ λ v,
+    by simp only [multilinear_map.coe_mk, smul_apply, ← sum.update_elim_inl, f.map_smul] }
+
+@[simp] lemma curry_sum_apply (f : multilinear_map R (λ x : ι ⊕ ι', M') M₂)
+  (u : ι → M') (v : ι' → M') :
+  f.curry_sum u v = f (sum.elim u v) :=
+rfl
+
+/-- A multilinear map on `Π i : ι, M'` taking values in the space of multilinear maps
+on `Π i : ι', M'` defines a multilinear map on `Π i : ι ⊕ ι', M'`. -/
+def uncurry_sum (f : multilinear_map R (λ x : ι, M') (multilinear_map R (λ x : ι', M') M₂)) :
+  multilinear_map R (λ x : ι ⊕ ι', M') M₂ :=
+{ to_fun := λ u, f (u ∘ sum.inl) (u ∘ sum.inr),
+  map_add' := λ u i x y, by cases i;
+    simp only [map_add, add_apply, sum.update_inl_comp_inl, sum.update_inl_comp_inr,
+      sum.update_inr_comp_inl, sum.update_inr_comp_inr],
+  map_smul' := λ u i c x, by cases i;
+    simp only [map_smul, smul_apply, sum.update_inl_comp_inl, sum.update_inl_comp_inr,
+      sum.update_inr_comp_inl, sum.update_inr_comp_inr] }
+
+@[simp] lemma uncurry_sum_aux_apply
+  (f : multilinear_map R (λ x : ι, M') (multilinear_map R (λ x : ι', M') M₂)) (u : ι ⊕ ι' → M') :
+  f.uncurry_sum u = f (u ∘ sum.inl) (u ∘ sum.inr) :=
+rfl
+
+variables (ι ι' R M₂ M')
+
+/-- Linear equivalence between the space of multilinear maps on `Π i : ι ⊕ ι', M'` and the space
+of multilinear maps on `Π i : ι, M'` taking values in the space of multilinear maps
+on `Π i : ι', M'`. -/
+def curry_sum_equiv : multilinear_map R (λ x : ι ⊕ ι', M') M₂ ≃ₗ[R]
+  multilinear_map R (λ x : ι, M') (multilinear_map R (λ x : ι', M') M₂) :=
+{ to_fun := curry_sum,
+  inv_fun := uncurry_sum,
+  left_inv := λ f, ext $ λ u, by simp,
+  right_inv := λ f, by { ext, simp },
+  map_add' := λ f g, by { ext, refl },
+  map_smul' := λ c f, by { ext, refl } }
+
+variables {ι ι' R M₂ M'}
+
+@[simp] lemma coe_curry_sum_equiv : ⇑(curry_sum_equiv R ι M₂ M' ι') = curry_sum := rfl
+
+@[simp] lemma coe_curr_sum_equiv_symm : ⇑(curry_sum_equiv R ι M₂ M' ι').symm = uncurry_sum := rfl
+
+variables (R M₂ M')
+
+/-- If `s : finset (fin n)` is a finite set of cardinality `k` and its complement has cardinality
+`l`, then the space of multilinear maps on `λ i : fin n, M'` is isomorphic to the space of
+multilinear maps on `λ i : fin k, M'` taking values in the space of multilinear maps
+on `λ i : fin l, M'`. -/
+def curry_fin_finset {k l n : ℕ} {s : finset (fin n)} [decidable_pred (s : set (fin n))]
+  (hk : s.card = k) (hl : sᶜ.card = l) :
+  multilinear_map R (λ x : fin n, M') M₂ ≃ₗ[R]
+    multilinear_map R (λ x : fin k, M') (multilinear_map R (λ x : fin l, M') M₂) :=
+(dom_dom_congr_linear_equiv M' M₂ R R (fin_sum_equiv_of_finset hk hl).symm).trans
+  (curry_sum_equiv R (fin k) M₂ M' (fin l))
+
+variables {R M₂ M'}
+
+lemma curry_fin_finset_apply {k l n : ℕ} {s : finset (fin n)} [decidable_pred (s : set (fin n))]
```

Comment:
This lemma and the next ones look like good simp lemmas. Is there a reason you did not put the attribute on them?

Reply:
Thanks, added several `simp` attrs.